mbed_tools_ci/utils/spdx_helpers.py

```diff
@@ -0,0 +1,663 @@
+        package.file_name = determine_spdx_value(self.name)
+        package.download_location = determine_spdx_value(None)
+        package.homepage = determine_spdx_value(self.url)
+        package.originator = Person(determine_spdx_value(self.author),
+                                    determine_spdx_value(self.author_email))
+        package.license_declared = License.from_identifier(
+            str(determine_spdx_value(self.licence)))
+        package.summary = determine_spdx_value(self.description)
+        package.description = NoAssert()
+        package.spdx_id = f'SPDXRef-{self.id}'
+        package.conc_lics = NoAssert()
+        package.add_lics_from_file(License.from_identifier(
+            str(determine_spdx_value(self.licence))))
+        files = self.get_spdx_files()
+        if files:
+            for file in files:
+                package.add_file(file.generate_spdx_file())
+        else:
+            # Has to generate a dummy file because of the following rule in SDK:
+            # - Package must have at least one file
+            package.add_file(SpdxFile(Path(UNKNOWN), self._project_root,
+                                      self.licence).generate_spdx_file())
+        return package
+
+
+class SpdxFile:
+    """SPDX File.
+
+    See https://spdx.org/spdx-specification-21-web-version#h.nmf14n
+    """
+
+    def __init__(self, path: Path, project_root: Path,
+                 package_licence: str) -> None:
+        """Constructor."""
+        self._path = path
+        self._project_root = project_root
+        self._package_licence = package_licence
+
+    @property
+    def path(self) -> Path:
+        """Gets the file path.
+
+        Returns:
+            the file path
+        """
+        return self._path
+
+    @property
+    def unix_relative_path(self) -> str:
+        """Gets the unix relative path.
+
+        Returns:
+            the file path
+        """
+        if str(self.path) == UNKNOWN:
+            return UNKNOWN
+        unix_path = str(self.path.relative_to(self._project_root)
+                        ).replace('\\', '/')
+        return f'./{unix_path}'
+
+    @property
+    def name(self) -> str:
+        """Gets the file name.
+
+        Returns:
+            the fine name
+        """
+        return self._path.name
+
+    @property
+    def id(self) -> str:
+        """Gets a unique identifier.
+
+        Returns:
+            a UUID
+        """
+        # Generates a unique Id based on the name of the file
+        return generate_uuid_based_on_str(self.unix_relative_path)
+
+    @property
+    def sha1_check_sum(self) -> str:
+        """Gets file SHA1 hash.
+
+        Returns:
+            corresponding hash
+        """
+        return determine_sha1_hash_of_file(self._path)
+
+    @property
+    def licence(self) -> str:
+        """Determines licence from file notice.
+
+        Returns:
+            file's licence
+        """
+        # TODO determine licence from file notice
```

Comment:
Why do we need this property if the functionality isn't implemented yet?

Reply:
The licence of a file is the same has the one defined as the package level unless overidden by a notice in the file. The property is necessary and currently default to the package licence which is what we currently have. However, I was just putting a note that the correct check is not fully implemented